esp-hal/src/lib.rs

```diff
@@ -143,15 +143,20 @@
 // MUST be the first module
 mod fmt;
 
+#[cfg(riscv)]
+#[doc(hidden)]
+pub use esp_riscv_rt::entry as __entry;
```

Comment:
Probably these should just go in the `__macro_implementation` module instead?

Reply:
or no .... we need them to be public even without the "unstable" feature

Reply:
in that case let's make `__macro_implementation` stable but `doc(hidden)`, I agree with Jesse that we should try and collect this stuff in one place to track it